src/Microsoft.Framework.DependencyInjection.Abstractions/ServiceProviderExtensions.cs

```diff
@@ -82,7 +82,14 @@ public static IEnumerable<T> GetRequiredServices<T>([NotNull] this IServiceProvi
         public static IEnumerable<object> GetRequiredServices([NotNull] this IServiceProvider provider, [NotNull] Type serviceType)
         {
             var genericEnumerable = typeof(IEnumerable<>).MakeGenericType(new [] { serviceType });
-            return (IEnumerable<object>)provider.GetRequiredService(genericEnumerable);
+            var providers = (IEnumerable<object>)provider.GetRequiredService(genericEnumerable);
+            
+            if (!providers.Any())
```

Comment:
I was wondering the same thing. But since the generic `GetRequiredServices<T>` also uses `Any()`, I figured it was okay.

We could already enumerate the `providers` collection, check if `Count > 0` and return that `List<object>`. That should avoid the multiple enumerations since it's already a list, right?

Perhaps @davidfowl can help?